src/datasets/summarizations/trend-weekly-comparison-average.summarization.ts

```diff
@@ -0,0 +1,41 @@
+
+import { Summary } from './types';
+import { TimeSeriesPoint } from '../metas/types';
+import { cacheSummaries } from './utils/commons';
+import {
+  groupPointsByXWeek,
+} from './utils/time-series';
+import { formatY } from '../../utils/formatters';
+
+export function queryFactory(points: TimeSeriesPoint[]) {
+  return cacheSummaries(() => {
+    // Only consider weeks with more than 3 days when creating summaries
+    // Weeks with 3 days or less are considered to belong to last/next 30 days

```

Comment:
Not sure I understand. Are weeks with < 3 days omitted in this summary OR used as a comparison "last/next/30 days"

Reply:
Weeks with < 3 days omitted in this summary. Any suggestions on improving wording and the comments?

Reply:
"Weeks with 3 days or less are considered to belong to the last/next 30 days and are omitted from the summaries."

What defines a week here? Is it possible that you omit 2 partial weeks (one in the beginning and one at the end)? Or if it is just by number of days, do you always omit the beginning or end?

Reply:
> Is it possible that you omit 2 partial weeks (one in the beginning and one at the end)?
Yes.

Considering weeks with <= 3 days may make the average of y-values in a week noisy and further harm the quality of the summaries. For 28-day data (GA data), there are always 4 valid weeks based on this rule. But for 30-day or 31-day data, there might be 4 or 5 weeks to create summaries.